sdk/azidentity/cache/linux_test.go

```diff
@@ -18,21 +18,19 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-var (
-	ctx              = context.Background()
-	storageAvailable = false
-)
+var ctx = context.Background()
 
 func TestMain(m *testing.M) {
-	err := tryKeyring()
-	storageAvailable = err == nil
+	// ensure the user running tests in CI is able to write cache files
+	if tmpdir, ok := os.LookupEnv("Agent.TempDirectory"); ok {
```

Comment:
Sorry @chlowell I misled you a bit. In the yaml you can use `$(Agent.TempDirectory)` to set new env variables or pass as a parameter. However in the environment it gets converted to `AGENT_TEMPDIRECTORY`.